packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -165,27 +203,18 @@ def _get_tx_data(
                 "send_raw_transaction unsuccessful! Insufficient funds."
             )
 
+        # Trigger repricing mechanism if we have succesfully detected the strategy price
+        # and we receive an underpriced status.
+        if rpc_status == RPCResponseStatus.UNDERPRICED and gas_price:
+            self.params.gas_price = gas_price
+
```

Comment:
I had to move this from the bottom to here, or then we would not trigger the repricing mechanism as the next IF returns even for UNDERPRICED status.

Reply:
We need to think if this is safe. Do we always have the information that you moved if `rpc_status != RPCResponseStatus.SUCCESS`?

Reply:
I meant nonce and digest. The latter has now been moved again.